openehr-rm/src/main/java/com/nedap/archie/rm/support/identification/ObjectVersionId.java

```diff
@@ -17,4 +21,53 @@ public ObjectVersionId() {
     public ObjectVersionId(String value) {
         super(value);
     }
+
+    @JsonIgnore
+    @XmlTransient
+    public UID getObjectId() {
+        return getRoot();
+    }
+
+    @JsonIgnore
+    @XmlTransient
+    public UID getCreatingSystemId() {
+        if (getExtension() == null)
+            // TODO: what kind of exception should be thrown here?
+            throw new IllegalArgumentException("Invalid OBJECT_VERSION_ID. Needs to have EXTENSION.");
```

Comment:
I think we need to add a validation, similar to those in this pull request: https://github.com/openEHR/archie/pull/227 . Then you can first run object validations, to ensure these exceptions will not happen.

I think that's to be done after this pull request. And we will still need these exceptions, I'll check which exception we should add here.

Reply:
Alright. So, adding methods like this one? https://github.com/openEHR/archie/pull/227/files#diff-2dd0caa542d5da60d98a3fd28c408078c9a29bed82fe3264c41ba5ae87c5b010R80

Keep me posted on the progress and if there's anything I can help with I'll try to do so.

Reply:
I will add these methods later, for now I think this exception is fine.

Reply:
Well, could be better since of course this is not an argument. Could be UnsupportedOperationException or simply NullPointerException? If not, we may have to define our own in Archie at some point.

Reply:
I did choose `IllegalArgumentException` because the actual problem happened during the creation of the object with an invalid argument. As I wrote in another comment, I think what I would do here would be to only allow the creation of valid objects. Maybe the validation you mentioned will be able to do that in the future.
But sure, for now I'll change the exceptions to `UnsupportedOperationException`.